contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,794 @@
+            // that stems from miscoordination between ytoken .deposit() amount wrapped and pricePerShare
+            // at that point in time.
+            // ex: if I have 1 eth, deposit 1 eth into yearn vault and calculate value of yearn token balance
+            // denominated in eth (via balance(yearn token) * pricePerShare) we will get 1 eth - 1 wei.
+            ISTETH(collateralToken).deposit(amountToWrap, address(this));
+        }
+    }
+
+    function getVaultFees(
+        Vault.VaultState storage vaultState,
+        uint256 currentLockedBalance,
+        uint256 performanceFeePercent,
+        uint256 managementFeePercent
+    )
+        external
+        view
+        returns (
+            uint256 performanceFee,
+            uint256 managementFee,
+            uint256 vaultFee
+        )
+    {
+        uint256 prevLockedAmount = vaultState.lastLockedAmount;
+        uint256 totalPending = vaultState.totalPending;
+
+        // Take performance fee and management fee ONLY if difference between
+        // last week and this week's vault deposits, taking into account pending
+        // deposits and withdrawals, is positive. If it is negative, last week's
+        // option expired ITM past breakeven, and the vault took a loss so we
+        // do not collect performance fee for last week
+        if (currentLockedBalance.sub(totalPending) > prevLockedAmount) {
+            performanceFee = currentLockedBalance
+                .sub(totalPending)
+                .sub(prevLockedAmount)
+                .mul(performanceFeePercent)
+                .div(100 * 10**6);
+            managementFee = currentLockedBalance.mul(managementFeePercent).div(
+                100 * 10**6
+            );
+
+            vaultFee = performanceFee.add(managementFee);
+        }
+    }
+
+    /**
+     * @notice Helper function to make either an ETH transfer or ERC20 transfer
+     * @param weth is the weth address
+     * @param asset is the vault asset address
+     * @param recipient is the receiving address
+     * @param amount is the transfer amount
+     */
+    function transferAsset(
+        address weth,
+        address asset,
+        address payable recipient,
+        uint256 amount
+    ) public {
+        if (asset == weth) {
+            IWETH(weth).withdraw(amount);
+            (bool success, ) = recipient.call{value: amount}("");
+            require(success, "!success");
+            return;
+        }
+        IERC20(asset).safeTransfer(recipient, amount);
+    }
+
+    /**
+     * @notice Returns the decimal shift between 18 decimals and asset tokens
+     * @param collateralToken is the address of the collateral token
+     */
+    function decimalShift(address collateralToken)
+        public
+        view
+        returns (uint256)
+    {
+        return
+            10**(uint256(18).sub(IERC20Detailed(collateralToken).decimals()));
```

Comment:
Given the fact that VaultLifecycleSTETH is only *limited* to stETH, we don't have to call `decimals()` at all and just 18

We can even remove the decimalShift function entirely because `10**(18-18) = 1`